jolt-evm-verifier/script/src/bin/hyperkzg_batch_example.rs

```diff
@@ -0,0 +1,173 @@
+use alloy_primitives::{hex, U256};
+use alloy_sol_types::{sol, SolType};
+
+use ark_bn254::Bn254;
+use ark_ec::pairing::Pairing;
+use ark_ff::BigInteger;
+use ark_ff::PrimeField;
+use ark_std::UniformRand;
+use jolt_core::poly::commitment::commitment_scheme::{BatchType, CommitmentScheme};
+use jolt_core::poly::commitment::hyperkzg::*;
+use jolt_core::poly::dense_mlpoly::DensePolynomial;
+use jolt_core::utils::transcript::ProofTranscript;
+use rand_chacha;
+use rand_core::SeedableRng;
+
+fn main() {
+    // Testing 2^12 ie 4096 elements
+    // We replicate the behavior of the standard rust tests, but output
+    // the proof and verification key to ensure it is verified in sol as well.
+
+    let ell = 12;
+    let mut rng = rand_chacha::ChaCha20Rng::seed_from_u64(ell as u64);
+
+    let n = 1 << ell; // n = 2^ell
+
+    let srs = HyperKZGSRS::setup(&mut rng, n);
+    let (pk, vk): (HyperKZGProverKey<Bn254>, HyperKZGVerifierKey<Bn254>) = srs.trim(n);
+
+    let point = (0..ell)
+        .map(|_| <Bn254 as Pairing>::ScalarField::rand(&mut rng))
+        .collect::<Vec<_>>();
+
+    let mut polys = vec![];
+    let mut evals = vec![];
+    let mut commitments = vec![];
+    let mut borrowed = vec![];
+    for _ in 0..8 {
+        let poly = DensePolynomial::new(
+            (0..n)
+                .map(|_| <Bn254 as Pairing>::ScalarField::rand(&mut rng))
+                .collect::<Vec<_>>(),
+        );
+        let eval = poly.evaluate(&point);
+        commitments.push(HyperKZG::commit(&pk, &poly).unwrap());
+        polys.push(poly);
+        evals.push(eval);
+    }
+
+    for poly in polys.iter() {
+        borrowed.push(poly);
+    }
+
+    // prove an evaluation
+    let mut prover_transcript = ProofTranscript::new(b"TestEval");
+    let proof: HyperKZGProof<Bn254> = HyperKZG::batch_prove(
```

Comment:
We likely will want to write this conversion logic directly on an `Into<HyperKZGSol>` trait eventually, but good for now.

Reply:
Done in the next pr, but good note.